modules/apigee_edge_teams/src/EventSubscriber/ValidateApigeeXTeamEnabledSubscriber.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * Copyright 2023 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_teams\EventSubscriber;
+
+use Drupal\apigee_edge\Entity\Controller\OrganizationControllerInterface;
+use Drupal\apigee_edge\SDKConnectorInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Validates that Apigee X Team is enabled on every Team page request.
+ */
+final class ValidateApigeeXTeamEnabledSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The SDK connector service.
+   *
+   * @var \Drupal\apigee_edge\SDKConnectorInterface
+   */
+  private SDKConnectorInterface $connector;
+
+  /**
+   * The organization controller service.
+   *
+   * @var \Drupal\apigee_edge\Entity\Controller\OrganizationControllerInterface
+   */
+  protected OrganizationControllerInterface $orgController;
+
+  /**
+   * The messenger service.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected MessengerInterface $messenger;
+
+  /**
+   * ValidateApigeeXTeamEnabledSubscriber constructor.
+   *
+   * @param \Drupal\apigee_edge\SDKConnectorInterface $sdk_connector
+   *   The SDK connector service.
+   * @param \Drupal\apigee_edge\Entity\Controller\OrganizationControllerInterface $org_controller
+   *   The organization controller service.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   The messenger service.
+   */
+  public function __construct(SDKConnectorInterface $sdk_connector, OrganizationControllerInterface $org_controller, MessengerInterface $messenger) {
+    $this->connector = $sdk_connector;
+    $this->orgController = $org_controller;
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * If monetization enabled in Apigee X org alert the user.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\RequestEvent $event
+   *   The event.
+   */
+  public function validateApigeexTeamEnabled(RequestEvent $event) {
+    /** @var \Symfony\Component\Routing\Route $current_route */
+    if (($current_route = $event->getRequest()->get('_route')) && (strpos($current_route, 'entity.team') !== FALSE || strpos($current_route, 'settings.team') !== FALSE)) {
```

Comment:
It just occurred to me, is this a message that everybody should see or only users with certain admin permission(s)?

Reply:
Also, the code still reacts to non-HTML requests if I am not mistaken.

Reply:
> It just occurred to me, is this a message that everybody should see or only users with certain admin permission(s)?

Everybody as of now.

Reply:
> Also, the code still reacts to non-HTML requests if I am not mistaken.

Done

Reply:
> >   It just occurred to me, is this a message that everybody should see or only users with certain admin permission(s)?

> Everybody as of now.

Well, I am skeptical if Apigee customers are going to like that their customers are going to see messages like this if they still keep using the Teams feature on their site - which seems to be working. 

My opinion that the goal should be reaching out the right audience and convincing them to disable this feature.

![image](https://github.com/apigee/apigee-edge-drupal/assets/1755573/5eb131e9-3473-4486-bafc-9907e503c355)
(The screenshot was made as a simple logged in user.)

Reply:
I agree with you. 
We can show different notice message for simple logged in user or no message at all for non-admin?

@giteshk What do you think?

Reply:
@giteshk and @mxr576 
As discussed, we will implement this only for admin and no error/notice message for non-admin users.

Reply:
Done